pkg/osbuild/device.go

```diff
@@ -242,21 +260,30 @@ func genOsbuildMount(name, source string, mnt disk.Mountable) (*Mount, error) {
 	}
 }
 
+// genMountsDevicesFromPt generates osbuild mounts and devices from a disk.PartitionTable
+// filename is the name of the underlying image file (which will get loop-mounted).
+//
+// Returned values:
+// 1) the name of the mount for the filesystem root
+// 2) generated mounts
+// 3) generated devices
+// 4) error if any
 func genMountsDevicesFromPt(filename string, pt *disk.PartitionTable) (string, []Mount, map[string]Device, error) {
 	devices := make(map[string]Device, len(pt.Partitions))
 	mounts := make([]Mount, 0, len(pt.Partitions))
 	var fsRootMntName string
 	genMounts := func(mnt disk.Mountable, path []disk.Entity) error {
-		stageDevices, name := getDevices(path, filename, false)
+		stageDevices, leafDeviceName := getDevices(path, filename, false)
```

Comment:
I wonder if we should take the opportunity to add a tiny bit more testing around `genMountsDevicesFromPt()` ? Not much just generate a few device names, something like:
```go
diff --git a/pkg/osbuild/device_test.go b/pkg/osbuild/device_test.go
index a8232bd2c..035dac4b2 100644
--- a/pkg/osbuild/device_test.go
+++ b/pkg/osbuild/device_test.go
@@ -188,12 +188,14 @@ func TestMountsDeviceFromPtNoRootErrors(t *testing.T) {
 
 func TestMountsDeviceFromPtHappy(t *testing.T) {
        filename := "fake-disk.img"
-       fakePt := testdisk.MakeFakePartitionTable("/")
+       fakePt := testdisk.MakeFakePartitionTable("/", "/boot", "/boot/efi")
        fsRootMntName, mounts, devices, err := genMountsDevicesFromPt(filename, fakePt)
        require.Nil(t, err)
        assert.Equal(t, fsRootMntName, "-")
        assert.Equal(t, mounts, []Mount{
                {Name: "-", Type: "org.osbuild.ext4", Source: "-", Target: "/"},
+               {Name: "boot", Type: "org.osbuild.ext4", Source: "boot", Target: "/boot"},
+               {Name: "boot-efi", Type: "org.osbuild.fat", Source: "boot-efi", Target: "/boot/efi"},
        })
        assert.Equal(t, devices, map[string]Device{
                "-": {
@@ -203,5 +205,19 @@ func TestMountsDeviceFromPtHappy(t *testing.T) {
                                Size:     testdisk.FakePartitionSize / 512,
                        },
                },
+               "boot": {
+                       Type: "org.osbuild.loopback",
+                       Options: &LoopbackDeviceOptions{
+                               Filename: "fake-disk.img",
+                               Size:     testdisk.FakePartitionSize / 512,
+                       },
+               },
+               "boot-efi": {
+                       Type: "org.osbuild.loopback",
+                       Options: &LoopbackDeviceOptions{
+                               Filename: "fake-disk.img",
+                               Size:     testdisk.FakePartitionSize / 512,
+                       },
+               },
        })
 }
``` 
maybe?

Reply:
Done! :)